src/types/user.ts

```diff
@@ -0,0 +1,17 @@
+export interface User {
+  firstName: string;
+  lastName: string;
+  email: string;
+  isPlatformAdmin: boolean;
+  id: number;
+  teams: UserTeams[];
+}
+
+export interface UserTeams {
+  teamId: number;
+  role: TeamRole;
+}
+
+export type TeamRole = "teamAdmin" | "teamViewer";
+
+export type Role = TeamRole | "public" | "platformAdmin";
```

Comment:
I'm curious if it's helpful via types to try to be more clear about what roles we've defined, versus which are Hasura defaults, for example: 
```ts
export type TeamRole = "teamAdmin" | "teamViewer";
export type HasuraRole = "public"; // "public" | "admin" ??
export type Role = TeamRole | HasuraRole | "platformAdmin"; 
```
We might not have enough roles to justify this yet, and it's still a tiny bit confusing here to communicate how `platformAdmin` is a permissions level rather than role, but just thinking ahead !

Reply:
This is a great idea - worth being as explicit as we can here in this fuzzy domain between database and logic.